packages/fs-detectors/src/detect-builders.ts

```diff
@@ -305,6 +305,22 @@ export async function detectBuilders(
     };
   }
 
+  // Exclude the middleware builder for Next.js apps since @vercel/next
+  // will build middlewares.
+  //
+  // While maybeGetApiBuilder() excludes the middleware builder, however,
+  // we need to check if it's a Next.js app here again for the case where
+  // `projectSettings.framework == null`.
+  if (frontendBuilder?.use === '@vercel/next' && apiBuilders.length > 0) {
```

Comment:
Lets use the same logic that is used to detect nextjs here since the project createdAt date matters 

https://github.com/vercel/vercel/blob/181a492d912b4c11970f824afc43638519a5ef7f/packages/fs-detectors/src/detect-builders.ts#L497

Reply:
In fact, you might be able to use `detectFrontBuilder()` instead